File: php/1346_Check_If_N_and_Its_Double_Exist.php
<?php

class Solution 
{

    /**
     * @param Integer[] $arr
     * @return Boolean
     */
    function checkIfExist($arr) 
    {
        $set = [];
        
        foreach ($arr as $num) {
            if (
                isset($set[$num * 2]) 
                || ($num % 2 == 0 && isset($set[$num / 2]))
            ) {
                return true;
            }

            $set[$num] = false;;
        }

        return false;
    }
}

File: go/1346_Check_If_N_and_Its_Double_Exist.go
func checkIfExist(arr []int) bool {
    set := make(map[int]bool) 
    for _, num := range arr {
        if set[num * 2] || (num % 2 == 0 && set[num / 2]) {
            return true;
        }

        set[num] = true;;
    }

    return false;
}
